src/validator.cpp

```diff
@@ -247,8 +277,38 @@ struct Validator::ValidatorImpl
      */
     void validateMathMLElements(const XmlNodePtr &node, const ComponentPtr &component);
 
+    /**
+     * @brief Validate and clean the @cn node.
+     *
+     * Validate the @c cn node and clear any CellML namespace from the node.
+     *
+     * @param node The node @c cn element.
+     * @param component The component the @p node is a part of.
+     */
     void validateAndCleanCnNode(const XmlNodePtr &node, const ComponentPtr &component) const;
+
+    /**
+     * @brief Validate that the @c ci node has a reference to a variable.
+     *
+     * Validate that the @c ci node has a reference to a variable.
+     *
+     * @param node The node @c ci element from the document.
+     * @param component The component the @p node is a part of.
+     * @param variableNames A list of variable names.
+     */
     void validateAndCleanCiNode(const XmlNodePtr &node, const ComponentPtr &component, const std::vector<std::string> &variableNames) const;
+
+    /**
+     * @brief Validate the text of a @c cn element.
+     *
+     * Validates that the @c cn element has a units attached.
+     *
+     * @param component The component that the @c cn element belongs to.
+     * @param unitsName The name of the units.
+     * @param textNode The text of the @c cn element.
+     *
+     * @return  @c true if the @c cn units is valid, @c false otherwise.
+     */
```

Comment:
Sometimes, we talk about `ci`/`cn` element and sometimes a `ci`/`cn` node. Would be good to be consistent.